sympy/solvers/bivariate.py

```diff
@@ -196,20 +241,74 @@ def _solve_lambert(f, symbol, gens):
       a = -1, d = a*log(p), f = -log(d) - g*log(p)
     """
 
+    def _solve_even_degree_expr(expr, symbol):
+        """
+        Helper function to consider two equations obtained if even degree
+        of symbol is present in the expr..
+
+        For example:
+        Equation (a/x + exp(x/2)).diff(x) = 0 is represented
+        in logform as both of the below equations.
+        1. 2*log(x) + x/2 - log(a) = 0
+        2. 2*log(-x) + x/2 - log(a) = 0
+
+        Parameters
+        ==========
+
+        expr : Expr
+            The expression which includes a dummy variable t which can be
+            replaced with +symbol and -symbol.
+
+        symbol : Symbol
+            The symbol concerned in expr.
+
+        Returns
+        =======
+
+        Combined solution after considering all even degree branches of
+        original equation.
+
+        """
+        llhs1, llhs2 = map(lambda i: expr.xreplace({t: i}), (symbol, -symbol))
+        if llhs1 == llhs2:
+            return _solve_lambert(llhs1, symbol, gens)
+        else:
+            sol1, sol2 = map(lambda i: _solve_lambert(i, symbol, gens), (llhs1, llhs2))
+            return list(set(sol1 + sol2))
+
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
+    even_degrees = [i for i in list(set(flatten(_power_list(lhs, symbol)))) \
+                    if i%2 == 0 and i != 0]
+    t = Dummy('t', **symbol.assumptions0)
+
     lamcheck = [tmp for tmp in gens
                 if (tmp.func in [exp, log] or
                 (tmp.is_Pow and symbol in tmp.exp.free_symbols))]
     if not lamcheck:
         raise NotImplementedError()
 
+    if lhs.is_Add and len(even_degrees) == 1:
+        lhs = lhs.xreplace({symbol**even_degrees[0]: t**even_degrees[0]})
+        if lhs.has(t):
+            other = lhs.subs(t, 0)
+            t_term = lhs - other
+            rhs = rhs - other
+            diff = expand_log(log(t_term) - log(rhs))
+            return _solve_even_degree_expr(diff, symbol)
+
     if lhs.is_Mul:
+        if len(even_degrees) != 0:
+            for i in even_degrees:
+                lhs = lhs.xreplace({symbol**i: t**i})
```

Comment:
`lhs.xreplace({symbol**i: t**i for i in even_degrees})` multiple calls to `xreplace` is expensive

Reply:
Thanks. Yeah I know i was putting `for i in even_degrees` outside of {}